packages/filebrowser/src/listing.ts

```diff
@@ -2660,13 +2660,24 @@ export namespace DirListing {
         checkbox.checked = selected ?? false;
       }
 
-      let modText = '';
+      let modHTML = '';
       let modTitle = '';
       if (model.last_modified) {
-        modText = Time.formatHuman(new Date(model.last_modified));
+        // Provide multiple formats, with container queries used to display exactly one
+        modHTML = ['narrow', 'short', 'long']
+          .map(
+            style =>
+              `<div class='${ITEM_MODIFIED_CLASS}-${style}'>` +
+              Time.formatHuman(
+                new Date(model.last_modified),
+                style as Time.HumanStyle
+              ) +
+              '</div>'
+          )
+          .join('');
         modTitle = Time.format(new Date(model.last_modified));
       }
-      modified.textContent = modText;
+      modified.innerHTML = modHTML;
```

Comment:
A solution in-between the re-write to an implementation that only ever renders one format, and this implementation which manually assembles the HTML code and forces it on the node (which will force-update the DOM each time resulting in bad performance), we could replace this fragment to use [`VirtualDOM.render`](https://lumino.readthedocs.io/en/latest/api/functions/virtualdom.VirtualDOM.render.html). This would mean that the DOM nodes are not touched unless there was a change.